Fixed mobile Safari issue causing media query event listener to fail.

ah-simonkrix opened the pull request:
In Safari on iOS, the media prop on the wrapped component isn't updating when the mobile device rotates.

The reason is that the listener is depending on the received event to be an object with a `target` property with the media query object in it, but that property isn't provided by mobile Safari.

Instead, it provides a much simpler object (`matches` and `media` are present - those are guaranteed by the spec)

jooj123 commented:
@ah-simonkrix this fails tests, also can you add a description as to why this fixes a mobile safari to fail ?

ah-simonkrix commented:
Sorry about the tests, I'll update the PR with a solution for that.

ah-simonkrix commented:
In order to fix the tests, I removed the dependency on package `match-media-mock` (which contains a bug), and replaced it with a small dummy function which just provides the media query matching and nothing else.

willmendesneto commented on src/media-query-provider.js:
```diff
@@ -56,13 +56,12 @@ class MediaQueryProvider extends React.Component {
-  mediaQueryListener({ matches, target }) {
-    const queryName = this.mediaQueryListInstanceMap.get(target);
+  mediaQueryListener({ matches, media }) {
+    const queryName = this.mediaQueryListInstanceMap.get(media).queryName;
```
`const { queryName } = this.mediaQueryListInstanceMap.get(media)`

albertstill commented:
There is a potential issue because the `media` query string is reformatted sometimes.

```js
var mql = window.matchMedia('(min-width: 500px) and (max-width: 600px)');

function screenTest(e) {
  console.log(e.media); // prints (max-width: 600px) and (min-width: 500px)
}

mql.addListener(screenTest);
``` 

Does Safari have `currentTarget` or `srcElement`?

![screen shot 2018-07-24 at 12 41 13 pm](https://user-images.githubusercontent.com/2787876/43113901-b1a1dd84-8f3f-11e8-912f-da368e2b1b0a.png)

albertstill commented on src/media-query-provider.js:
```diff
@@ -56,13 +56,12 @@ class MediaQueryProvider extends React.Component {
-    this.mediaQueryListInstanceMap.forEach((_, mediaQueryList) => {
-      mediaQueryList.removeListener(this.mediaQueryListener);
-    });
+    this.mediaQueryListInstanceMap.forEach(mediaQueryList =>
```
I think this won't work, you need the second arg for the value so will have to keep the `(_, mediaQueryList)`

albertstill replied:
Ah my bad, nice one

ah-simonkrix commented:
Unfortunately the event object that Safari passes through doesn't have any values apart from `media` and `matches` (as well as the object's methods).

ah-simonkrix commented:
Here's the event passed in:
![screen shot 2018-07-24 at 1 35 24 pm](https://user-images.githubusercontent.com/17535366/43115257-736d6a0e-8f46-11e8-87d0-ff64b5346eee.png)

albertstill commented:
nice work mate, so strange Safari gives you a `MediaQueryList` and Chrome gives you a `MediaQueryListEvent`

jooj123 commented:
LGTM - merging and releasing 🎉 - thanks @ah-simonkrix

ah-simonkrix commented:
Thank you very much for your help guys!
